### PR TITLE
TAO-10473 Fixes for new metadata discovery endpoint

### PR DESCRIPTION
**Pull request by luisfn**
This PR is intended to fix issues found during the development of its frontend counterpart

**Note**: MetadataCollection.php is not a new file, it was just renamed due to case sensitiveness issue

### CODE REVIEW

**Review by Skyld — CHANGES_REQUESTED** [review]
Could you please add tests?

**Comment by luisfn** [comment]
> Could you please add tests?

There are tests already for these. For the "new" class, it is not new, it was just renamed, not sure it was not recognized as such

**Review by andreluizmachado — CHANGES_REQUESTED** [review]

File: models/classes/Lists/Business/Domain/MetadataCollection.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+use Traversable;
+
+class MetadataCollection implements IteratorAggregate, JsonSerializable, Countable
+{
+    /** @var Metadata[] */
+    private $items = [];
+
+    public function __construct(Metadata ...$items)
+    {
+        foreach ($items as $value) {
```
[the remark this answers is not in the document]
Comment (luisfn) [reply models/classes/Lists/Business/Domain/MetadataCollection.php:38]: Thanks for the suggestion. Did the same for the other collection as well

**Review by andreluizmachado — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by Skyld — APPROVED** [review]

**Pull request merged**